toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/Map.kt

```diff
@@ -52,6 +53,94 @@ public fun Map(modifier: Modifier = Modifier, mapState: MapState = MapState()) {
                 mapView.map = it
             }
         }
+
+        // Collect set viewpoint operation and assign the passed value to the MapView
+        launch {
+            mapState.setViewpointChannel.receiveAsFlow().collect {
+                when (it) {
+                    is SetViewpointOperation.SetViewpoint -> {
+                        mapView.setViewpoint(it.viewpoint)
+                    }
+                }
+            }
+        }
+
+        // Collect get current viewpoint operation and pass the current viewpoint back to the channel
+        launch {
+            mapState.getCurrentViewpointChannel.receiveAsFlow().collect {
+                when (it) {
+                    is GetCurrentViewpointOperation.GetCurrentViewpoint -> {
+                        val currentViewpoint = mapView.getCurrentViewpoint(it.viewpointType)
+                        if (currentViewpoint != null) {
+                            it.completeWith(Result.success(currentViewpoint))
+                        } else {
+                            it.completeWith(Result.failure(Throwable("Error retrieving the current viewpoint")))
```

Comment:
I think returning a failure result in case of a null viewpoint makes sense, users can then call `result.getOrNull()`. I just think using a more specific exception type would make sense:
```suggestion
                            it.completeWith(Result.failure(IllegalStateException("Unable to retrieve the current viewpoint")))
```